internal/filter/filter/testdata/config.yaml

```diff
@@ -0,0 +1,16 @@
+# Yaml form of the configuration for FilterSets
+# This configuration can be embedded into other component's yamls
+# The top level here are just test names and do not represent part of the actual configuration.
+
+regexp/default:
+    match_type: regexp
```

Comment:
Why `match_type`? I thought we implement the Option B from https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161